test/t8_cmesh/t8_gtest_cmesh_face_is_boundary.cxx

```diff
@@ -0,0 +1,188 @@
+    cmesh = t8_cmesh_new_from_class (eclass, sc_MPI_COMM_WORLD);
+  }
+  void TearDown() override {
+    t8_cmesh_destroy(&cmesh);
+  }
+
+  t8_cmesh_t          cmesh;
+  t8_eclass           eclass;
+  int                 num_faces;
+  int                 iface;
+};
+
+TEST_P (cmesh_face_boundary_one_tree, check_face_is_boundary_one_tree) {
+  
+    EXPECT_TRUE(t8_cmesh_is_committed (cmesh));
+    /* We now check each face */
+    num_faces = t8_eclass_num_faces[(int) eclass];
+    for (iface = 0; iface < num_faces; ++iface) {
+      EXPECT_TRUE(t8_cmesh_tree_face_is_boundary (cmesh, 0, iface));
+      EXPECT_TRUE(t8_cmesh_get_face_neighbor (cmesh, 0, iface, NULL, NULL) < 0);
+    }
+}
+
+INSTANTIATE_TEST_SUITE_P(t8_gtest_cmesh_face_boundary, cmesh_face_boundary_one_tree, testing::Range(T8_ECLASS_ZERO, T8_ECLASS_COUNT));
+
+/* For a two tree cmesh, compute a parallel distribution of the trees.
+ * If we have more than 1 process, the first half of process (rank < size/2)
+ * gets tree 0, the other half gets tree 1. */
+static void
+t8_test_compute_parallel_bounds (sc_MPI_Comm comm, t8_gloidx_t *first_tree,
+                                 t8_gloidx_t *last_tree)
+{
+  int                 mpirank, mpisize, mpiret;
+  int                 first_tree_shared = 0;
+
+  mpiret = sc_MPI_Comm_rank (comm, &mpirank);
+  SC_CHECK_MPI (mpiret);
+  mpiret = sc_MPI_Comm_size (comm, &mpisize);
+  SC_CHECK_MPI (mpiret);
+
+  /* If only one process, it gets all trees */
+  if (mpisize == 1) {
+    *first_tree = 0;
+    *last_tree = 1;
+    return;
+  }
+  /* First half of processes gets tree 0, other half gets tree 1 */
+  if (mpirank < mpisize / 2) {
+    /* The first tree is shared if this rank is in the lower half, but not rank 0 */
+    first_tree_shared = mpirank > 0;
+    /* The first tree is 0 */
+    *first_tree = first_tree_shared ? -1 : 0;
+    *last_tree = 0;
+  }
+  else {
+    /* The first tree is shared if this process is not mpisize/2 */
+    first_tree_shared = mpirank > mpisize / 2;
+    /* The first tree is 1 */
+    *first_tree = first_tree_shared ? -2 : 1;
+    *last_tree = 1;
+  }
+}
+
+/* Test for two tree cmesh
+ * Creates coarse meshes with two trees for each eclass,
+ * one for each face of the first tree as the connecting face.
+ * This, only the remaining trees should register as boundary trees. */
+class cmesh_face_boundary_two_trees : public testing::TestWithParam<std::tuple<t8_eclass,int>>{
+protected:
+  void SetUp() override {
+    eclass = std::get<0>(GetParam());
+    do_partition = std::get<1>(GetParam());
+    num_faces = t8_eclass_num_faces[(int)eclass];
+    t8_test_compute_parallel_bounds (sc_MPI_COMM_WORLD, &first_tree, &last_tree);
+
+  }
+
+  t8_cmesh_t          cmesh;
+
+  t8_eclass           eclass;
+  int                 do_partition;
+  int                 num_faces;
+  int                 iface;
+  int                 checkface;
+  t8_gloidx_t         first_tree;
+  t8_gloidx_t         last_tree;
```

Comment:
Please move all variables, that can not be initialized during SetUp out of the class. They are helper-variables for the test and it is ok to declare them there. We could write another test using this class which does not need ```checkface``` for example.